flink-table/flink-table-planner-blink/src/test/scala/org/apache/flink/table/planner/plan/stream/sql/join/LookupJoinTest.scala

```diff
@@ -211,24 +212,26 @@ class LookupJoinTest(legacyTableSource: Boolean) extends TableTestBase with Seri
     expectExceptionThrown(
       "SELECT * FROM T JOIN LookupTable7 " +
         "FOR SYSTEM_TIME AS OF T.proctime AS D ON T.a = D.id AND T.b = D.name AND T.ts = D.ts",
-      "Expected: eval(java.util.concurrent.CompletableFuture, " +
-        "java.lang.Integer, org.apache.flink.table.data.StringData, " +
-        "org.apache.flink.table.data.TimestampData) \n" +
-        "Actual: eval(java.lang.Integer, org.apache.flink.table.data.StringData, " +
-        "java.time.LocalDateTime)",
-      classOf[TableException]
+      "Could not find an implementation method 'eval' in class " +
+          "'org.apache.flink.table.planner.plan.utils.InvalidAsyncTableFunctionEvalSignature1' " +
+          "for function 'default_catalog.default_database.LookupTable7' that matches the " +
+          "following signature:\n" +
+          "void eval(java.util.concurrent.CompletableFuture, java.lang.Integer, " +
+          "org.apache.flink.table.data.StringData, org.apache.flink.table.data.TimestampData)",
```

Comment:
Do we always fail with the internal data types now? Could we fail with the default classes?

Reply:
No, we fail with internal classes if the function declared `RowData`. I changed some tests to reflect this.

Reply:
Thanks for clarifying it !